### PR TITLE
ghc: Update to version 9.6.4

### PR DESCRIPTION
**Pull request by essandess**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: lang/ghc/Portfile [hunk 1]
```diff
@@ -350,10 +364,15 @@ subport ghc-prebuilt {
...
+                    port:ld64 \
+                    port:llvm-${ghc_llvm_version}
+
     extract.only    ${ghc_distname}${extract.suffix}
-if {[exists extract.rename]} {
-    extract.rename  yes
-}
+    if {[exists extract.rename]} {
+        extract.rename  yes
+    }
```
Comment (ryandesign) [inline lang/ghc/Portfile:375]: The reason it was indented that way was that `if {[exists extract.rename]} {` and `}` were intended to be temporary, only needed during the brief transition period leading up to and shortly after the January 2023 release of MacPorts 2.8.1 which introduced the `extract.rename` option. Now that that release has been out for well over two weeks (a year, in fact), the `if` statement should be removed, leaving only the unconditional `extract.rename yes`.
```suggestion
    extract.rename  yes
```

Comment (essandess) [reply lang/ghc/Portfile]: Thanks. I implemented this, along with some accompanying improvements to PG `haskell_cabal`.

**Comment by essandess** [comment]
@ryandesign This is ready to merge, and preferable to merge prior to `cabal` PG-based  ports like #22524 and #22525.

File: _resources/port1.0/group/haskell_cabal-1.0.tcl [hunk 2]
```diff
@@ -42,7 +42,10 @@ proc haskell_cabal.add_dependencies {} {
             port:libiconv
     }
     depends_build-append \
-        port:gsed
+        port:cctools \
+        port:file \
+        port:gsed \
+        path:bin/openssl:openssl
```
Comment (reneeotten) [inline _resources/port1.0/group/haskell_cabal-1.0.tcl:48]: I am not sure this is what we want. Do you mean that an old "openssl" installed by some other means is okay to satisfy the dependency? Why not just depend unconditionally on it?

Comment (essandess) [reply _resources/port1.0/group/haskell_cabal-1.0.tcl:48]: Any `openssl` will do, and I have followed the practice here of other ports that specify the dependency `path:bin/openssl:openssl`.

That said, the intent here is to determine if two files differ within `tcl` (at about line 334 below), and I'm using a RIPEMD hash for that computes by `openssl` exactly as in [macports.tcl](https://github.com/macports/macports-base/blob/49df406d8273f203d30186a77251a897f9d8ef79/src/macports1.0/macports.tcl#L2970).

This is a simple way to accomplish file comparison, but I'm open to a better solution, perhaps just using `diff`.

**Pull request merged**